source/host/System.cc

```diff
@@ -11,6 +11,8 @@ namespace Host {
 /// @brief The main entry point for the program.
 /// @addr{0x80008EF0}
 int KSystem::main(int argc, char **argv) {
+    EGG::Heap::initialize();
```

Comment:
This function call must happen before any allocations can be made. Otherwise, the current heap is set to nullptr, and the program segfaults.

In the base game, `RKSystem::main` calls `RKSystem::initialize`, which calls `RKSystem::initMemory`. `initMemory` is the function responsible for calling `EGG::Heap::initialize` and creating heaps out of the arenas. However, the base game also doesn't handle options until much later. For context, see 8000a8f0, and examine where it gets called from `RKSystem::initialize` - it's after.

Rather than overhaul the option parsing code (which we will need to do later), I instead opt for the simpler approach of just initializing memory immediately in `KSystem::main`.